Tweak.xm

```diff
@@ -1,173 +1,129 @@
-            CGRect origFrame = arg1;
-            CGRect fixedFrame = CGRectMake(0, 50, origFrame.size.width, origFrame.size.height);
-            %orig(fixedFrame);
-        } else {
-            %orig;
-        }
-    } else {
-		%orig;
-	}
+
+    CGFloat dockHeight = kRootFolderController.dockHeight;
+    if ([self.iconLocation containsString:@"Root"])
+        self.additionalLayoutInsets = UIEdgeInsetsMake(dockHeight,0,-dockHeight,0);
+
+    kDockView.hidden = shouldHide;
+
+    return x;
 }
 
--(void)initWithFrame:(CGRect)arg1{
-    if (shouldHide) {
-        [self setHidden:TRUE];
-    }
-    if (@available(ios 13, *)) {
-        UIEdgeInsets safeAreaInsets = [[[UIApplication sharedApplication] keyWindow] safeAreaInsets];
-        if (safeAreaInsets.bottom != 0){
-            CGRect origFrame = arg1;
-            CGRect fixedFrame = CGRectMake(0, 50, origFrame.size.width, origFrame.size.height);
-            %orig(fixedFrame);
-        } else {
-            %orig;
-        }
-    } else {
-		%orig;
-	}
+%end
+
+%hook SBDockView 
+
+-(void)setHidden:(BOOL)arg 
+{
+    %orig(shouldHide ? YES : arg);
 }
 
--(CGPoint)center{
-    if (shouldHide) {
-        [self setHidden:TRUE];
-    }
-    if (@available(ios 13, *)) {
-        UIEdgeInsets safeAreaInsets = [[[UIApplication sharedApplication] keyWindow] safeAreaInsets];
-        if (safeAreaInsets.bottom != 0){
-            if (!dockPoint.x) {
-                CGPoint origPoint = %orig;
-                dockPoint = CGPointMake(origPoint.x, (origPoint.y + 50));
-            }
-            return dockPoint;
-        } else {
-            return %orig;
-        }
-    } else {
-		return %orig;
-	}
+-(BOOL)isHidden
+{
+    return shouldHide ? YES : %orig;
 }
 
 -(void)setCenter:(CGPoint)arg1{
-    if (shouldHide) {
-        [self setHidden:TRUE];
+    if (shouldHide)
+    {
+        // %orig(CGPointMake(arg1.x, (arg1.y - 1000))); (maybe better? less objc calls)
+        self.hidden = YES;
+        return;
     }
-    if (@available(iOS 13, *)) {
-        UIEdgeInsets safeAreaInsets = [[[UIApplication sharedApplication] keyWindow] safeAreaInsets];
-        if (safeAreaInsets.bottom != 0){
-            if (!dockPoint.x) {
-                CGPoint origPoint = arg1;
-                dockPoint = CGPointMake(origPoint.x, (origPoint.y + 50));
-            }
-            %orig(dockPoint);
-        } else {
-            %orig;
-        }
-    } else {
-		%orig;
-	}
-}
-
--(void)setHidden:(BOOL)arg1{
-    if (shouldHide) {
-        %orig(TRUE);
-    } else {
-        %orig;
+    if (@available(ios 13, *) && safeAreaInsets.bottom != 0 && isEnabled) 
+    {
+        %orig(CGPointMake(arg1.x, (arg1.y + 50)));
+        return;
     }
+    %orig;
 }
 
 %end
 
-%end
-
-%group old
-
-%hook SBRootFolderView
+%hook SBIconModel
 
--(void)_coverSheetWillDismiss:(id) arg1{
-	self.dockEdge = 1;
-	%orig;
+-(void)layout
+{
+    %orig;
+    
+    [kDockView.superview bringSubviewToFront:kDockView];
```

Comment:
Whenever we destroy and rebuild the icon listing, we need to move the dock back to the front again.